### ISSUE TITLE
Tracking job completion in HTCondor

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#8488]
As discussed with Brian in https://github.com/dmwm/WMCore/pull/8469 (see full discussion over there), we can make some improvements to the job tracking where we can guarantee that the TransferOutput files have been transferred back to the schedd and that the job is really gone from the condor queue.

In short:
* only complete jobs that have left the condor queue
   * thus, do not complete jobs in `Completed` state. It should almost never happen, if it does, job will stay in this state for less than a sec
   * do not complete jobs in `Suspended` state (Error). It can go from suspended to pretty much any other state, so let's just ignore it.
   * do not complete jobs in `Removed` state (Error). It's a transient state and the job should leave the queue soon.
   * do not complete jobs in `Held` state (Error). It might stay here forever, but since we have a 10min PeridiocRemove expression, we can just wait for it to be gone from the queue. 

see: https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/BossAir/Plugins/SimpleCondorPlugin.py#L236

@bbockelm @ticoann feel free to complement anything that I might have missed.

**Comment by bbockelm** [comment]
Just wanted to note that `Completed` state always _happens_, but it's typically a fraction of a second so it is rarely _observed_.  It's only when the schedd is very busy that a backlog forms for completing jobs and you see many jobs in this state.

**Comment by amaltaro** [comment]
I didn't want to pollute this thread, but about "Just wanted to note that Completed state always happens". It happens only for jobs that end "gracefully", right? Or even Held/Suspended jobs still sit in completed for a fraction of a second before leaving the queue (like, Completed is always the last job state)

**Comment by amaltaro** [comment]
@bbockelm I don't remember the whole discussion. But if you can answer the question above, that will help once we resume working on this issue (which we know not to be optimal in the agent). Thanks

**Comment by bbockelm** [comment]
Sure (apologies, obviously missed the question first time around).

I do not know jobs that are removed from the queue go through a completed state transition.  However, since this state is typically only visible for a fraction of a second (sometimes not at all), state tracking code needs to gracefully handle the case where it doesn't observe this state -- i.e., it appears to go straight from "Held" to not in the queue.

**Comment by bbockelm** [comment]
> Comment from Todor: This was tried in CRAB and it didn’t scale well. QEDIT added load to the Schedd to the point it crashed.

This is incorrect.  When CRAB tried this, the initial PR actually changed two unrelated things.  One was doing the QEDIT approach, the other introduced a denial-of-service attack against the schedd.  Once we figured out what was going wrong and removed the DoS issue, qedit was fine.
